modules/core/04-channel/keeper/upgrade.go

```diff
@@ -596,6 +596,94 @@ func (k Keeper) startFlushUpgradeHandshake(
 	return nil
 }
 
+// ChanUpgradeOpen is called by a module to complete the channel upgrade handshake and move the channel back to an OPEN state.
+// This method should only be called after both channels have flushed any in-flight packets.
+func (k Keeper) ChanUpgradeOpen(
```

Comment:
there was a comment during the last upgradability call we did about a race condition between this open and the start of a new handshake. I don't believe there's much to do about it here.